src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Controllers/DraftApprenticeshipController.cs

```diff
@@ -279,6 +279,42 @@ public async Task<IActionResult> AddDraftApprenticeship(string changeCourse, str
             }
         }
 
+        [HttpGet]
+        [Route("add/apprenticeship/overlap-alert")]
+        public IActionResult DraftApprenticeshipOverlapAlert(DraftApprenticeshipOverlapAlertRequest request, [FromServices] IFeatureTogglesService<ProviderFeatureToggle> featureTogglesService)
+        {
+            var featureToggleEnabled = featureTogglesService.GetFeatureToggle(ProviderFeature.OverlappingTrainingDate).IsEnabled;
+
+            DraftApprenticeshipViewModel model;
+            if (request.DraftApprenticeshipHashedId != null)
+            {
+                model = PeekStoredEditDraftApprenticeshipState();
+            }
+            else
+            {
+                model = PeekStoredAddDraftApprenticeshipState();
+            }
+
+            var vm = new DraftApprenticeshipOverlapAlertViewModel
+            {
+                OverlappingTrainingDateRequestToggleEnabled = featureToggleEnabled,
+                DraftApprenticeshipHashedId = request.DraftApprenticeshipHashedId,
+                StartDate = model is null ? DateTime.MinValue : model.StartDate.Date.GetValueOrDefault(),
+                EndDate = model is null ? DateTime.MaxValue : model.EndDate.Date.GetValueOrDefault(),
+                Uln = model is null ? string.Empty : model.Uln,
+                FirstName = model is null ? string.Empty : model.FirstName,
+                LastName = model is null ? string.Empty : model.LastName
```

Comment:
same as my other comment about the model being null

Reply:
This has been updated. Unit tests updated accordingly.